### DISCUSSION TITLE
Adding a form in backend controller

### DISCUSSION THREAD

**Discussion by WebVPF** [wintercms/winter#565] [Q&A]
Please tell me what am I doing wrong.
I am writing a TODO Board plugin. Can't add form to add card description.
When I make a form I get an error:

> "A widget with class name 'formDesc' has not been bound to the controller" on line 606 of www\webvpf\modules\backend\classes\Controller.php

Form Field Definition File https://github.com/WebVPF/wn-todoboard-plugin/blob/c4c7bc7714935730f3baf0857002d4741044c6ad/controllers/board/form-desc-card.yaml

Form https://github.com/WebVPF/wn-todoboard-plugin/blob/c4c7bc7714935730f3baf0857002d4741044c6ad/controllers/board/_card-desc-form.htm

Visually the form looks correct

![2022-05-28_181350](https://user-images.githubusercontent.com/61043464/170831855-cb55aeab-c982-42c5-8c64-adaf2464431e.jpg)

Form widget code:
```php
public function onLoadFormCardDesc() {
    $card = Card::find(Input::get('id'));

    $config = $this->makeConfig($this->form);
    $config->model = $card;
    $widget = $this->makeWidget('Backend\Widgets\Form', $config);

    $this->vars['card'] = $card;
    $this->vars['widget'] = $widget;
}
```
https://github.com/WebVPF/wn-todoboard-plugin/blob/c4c7bc7714935730f3baf0857002d4741044c6ad/controllers/Board.php#L101-L117

**Comment by mjauvin** [comment]
https://wintercms.com/docs/backend/widgets#generic-binding
